dev/tasks/run-e2e

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+cd ${REPO_ROOT}
+
+export KUBEBUILDER_ASSETS=$(go run sigs.k8s.io/controller-runtime/tools/setup-envtest@latest use -p path)
+
+if [[ -z "${RUN_TESTS:-}" ]]; then
+  RUN_TESTS=TestAllInSeries/fixtures
+fi
+echo "Running tests matching: ${RUN_TESTS}"
+
+if [[ -z "${E2E_GCP_TARGET:-}" ]]; then
+  E2E_GCP_TARGET=mock
+fi
+echo "Using E2E_GCP_TARGET: ${E2E_GCP_TARGET}"
+export E2E_GCP_TARGET
+
+if [[ -z "${E2E_KUBE_TARGET:-}" ]]; then
+  E2E_KUBE_TARGET=envtest
+fi
+echo "Using E2E_KUBE_TARGET: ${E2E_KUBE_TARGET}"
+export E2E_KUBE_TARGET
+
+if [[ -z "${ARTIFACTS:-}" ]]; then
+  ARTIFACTS=${REPO_ROOT}/artifactz/${E2E_GCP_TARGET}
+fi
+echo "Using artifacts: ${ARTIFACTS}"
+export ARTIFACTS
+
+if [[ "${E2E_GCP_TARGET}" == "real" ]]; then
+  if [[ -z "${GCP_PROJECT_ID:-}" ]]; then
+    GCP_PROJECT_ID=$(gcloud config get-value project)
+  fi
+  echo "Using project: ${GCP_PROJECT_ID}"
+
+  TEST_FOLDER_ID=$(gcloud projects describe ${GCP_PROJECT_ID} --format='value(parent.id)')
+  export TEST_FOLDER_ID
+fi
```

Comment:
This will not work in GH actions so we probably want to use `secrets` in the yaml wf definition if we want to run these in GH actions eventually.

Reply:
True.  I don't think E2E_GCP_TARGET=real will work in GitHub actions.